backend/src/packages/users/user.service.ts

```diff
@@ -98,6 +102,33 @@ class UserService implements Service {
   public delete(): ReturnType<Service['delete']> {
     return Promise.resolve(true);
   }
+
+  public async findByEmail(
+    email: string,
+  ): Promise<ReturnType<UserEntity['toObject']> | null> {
+    const userEntity = await this.userRepository.findByEmail(email);
+
+    if (!userEntity) {
+      return null;
+    }
+
+    return userEntity.toObject();
+  }
+
+  public async findByEmailWithPassword(
+    email: string,
+  ): Promise<ReturnType<UserWithPasswordEntity['toObject']>> {
+    const userEntity = await this.userRepository.findByEmailWithPassword(email);
+
+    if (!userEntity) {
+      throw new UsersError({
+        status: HTTPCode.NOT_FOUND,
+        message: ExceptionMessage.USER_NOT_FOUND,
+      });
```

Comment:
let's return null here and throw this error in AuthService where we call this method

Reply:
Done!